### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by GigiDev7**
MongoDB/Mongoose - continue to add more logic

### CODE REVIEW

**Comment by GigiDev7** [comment]
finished accounts part, if all is good i'll megre it and continue with Transactions

**Comment by GigiDev7** [comment]
can i ask you one question?
nearly how many developers work on one project in a team?

File: controllers/transactions.js [hunk 1]
```diff
@@ -1,26 +1,66 @@
...
+  try {
+    const { accountId } = req.params;
+    const { category } = req.query;
+    let transactions;
+    if (category) {
+      transactions = await findTransactionsByCategory(accountId, category);
+      return res.status(200).json(transactions);
+    }
+    transactions = await findAllTransactions(accountId);
+    res.status(200).json(transactions);
```
Comment (Lokki17) [inline controllers/transactions.js:24]:
```
    let transactions;
    if (category) {
      transactions = await findTransactionsByCategory(accountId, category);
      return res.status(200).json(transactions);
    }
    transactions = await findAllTransactions(accountId);
    res.status(200).json(transactions);
```
It's easyer to read my code:

```
    let transactions;
    if (category) {
      transactions = await findTransactionsByCategory(accountId, category);
    } else {
      transactions = await findAllTransactions(accountId);
    }
    res.status(200).json(transactions);
```

**Comment by GigiDev7** [comment]
I have one more question, in app i have also category which has 'title' and 'type' properties and user can delete,update,create them, should i add seperate schema for that? or add categories array in user schema and this array will be full of objects with 'title' and 'type' properties?

**Comment by GigiDev7** [comment]
Backend part is basically finished, it may need some minor changes after we go to frontend part but it won't be problem

File: controllers/transactions.js [hunk 2]
```diff
@@ -1,26 +1,70 @@
...
+    const { category, type } = req.query;
+    let transactions;
+    if (category && type) {
+      transactions = await findTransactionsByCategory(
+        accountId,
+        category,
+        type
+      );
+      return res.status(200).json(transactions);
+    }
```
Comment (Lokki17) [inline controllers/transactions.js:22]: Why dont you want to use if - else here?

**Review by Lokki17 — APPROVED** [review]
Fixe remarks (up to you) and merge

**Comment by Lokki17** [comment]
It depeods on project. Sometimes it can be small group (3-5 persons) but sometimes it can be 4-5 groups (each has 8-10 persons) and even more

File: controllers/transactions.js (hunk 2 above)
Comment (GigiDev7) [reply controllers/transactions.js:22]: I got used to this kind of syntax some months ago, in lots of new code snippets i see less and less 'else' statements

Comment (GigiDev7) [reply controllers/transactions.js:22]: honestly i don't know why  :d

**Comment by GigiDev7** [comment]
It was nice collaboration. thank you

**Pull request merged**